app/controllers/concerns/course_basic_detail_concern.rb

```diff
@@ -89,18 +104,40 @@ def build_previous_course_creation_params
   end
 
   def next_step(current_step:)
-    if current_step == :outcome
-      new_provider_recruitment_cycle_courses_entry_requirements_path(
-        params[:provider_code],
-        params[:recruitment_cycle_year],
-        course: course_params,
-      )
-    elsif current_step == :entry_requirements
-      new_provider_recruitment_cycle_courses_outcome_path(
-        params[:provider_code],
-        params[:recruitment_cycle_year],
-        course: course_params,
-      )
+    next_step = NextCourseCreationStepService.new.execute(current_step: current_step)
+    next_page = course_creation_path_for(next_step)
+
+    if next_page.nil?
+      raise "No path defined for next step: #{next_step}"
+    end
+
+    next_page
+  end
+
+  def course_creation_path_for(page)
```

Comment:
can this be collapsed together...
they both got the giant case statement

service takes current and return next, and method takes next (page) and returns path